### PR TITLE
port gen-bbconfig to click and create a unit tests for the command

### PR DESCRIPTION
**Pull request by Sinerum**

### CODE REVIEW

File: tests/tools/test_driver_gen_benchbuild_config.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+    @run_in_test_environment()
+    def test_gen_bbconfig(self):
+        """basic tests for the `gen-bbconfig` command."""
+        runner = CliRunner()
+        runner.invoke(driver_gen_benchbuild_config.main, [])
+        self.assertTrue(
+            Path(vara_cfg()["benchbuild_root"].value +
+                 "/.benchbuild.yml").exists()
+        )
```
Comment (boehmseb) [inline tests/tools/test_driver_gen_benchbuild_config.py:24]: Have you checked that the file is not present in the beginning? If i remember correctly, the `@run_in_test_environment()` does create fresh configs.

File: tests/tools/test_driver_gen_benchbuild_config.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+class TestDriverGenBenchbuildConfig(unittest.TestCase):
+    """Tests for the driver_gen_benchbuild_config module."""
+
+    @run_in_test_environment()
+    def test_gen_bbconfig(self):
+        """basic tests for the `gen-bbconfig` command."""
+        runner = CliRunner()
+        os.remove(
+            Path(vara_cfg()["benchbuild_root"].value + "/.benchbuild.yml")
+        )
```
Comment (boehmseb) [inline tests/tools/test_driver_gen_benchbuild_config.py]: Prefer using `pathlib` over `os`. You can simply use [`Path.unlink()`](https://docs.python.org/3/library/pathlib.html#pathlib.Path.unlink) to remove a file.

**Review by boehmseb — CHANGES_REQUESTED** [review]

**Review by vulder — APPROVED** [review]
lgtm

Is there an issue for this PR?

**Comment by boehmseb** [comment]
> lgtm
> 
> Is there an issue for this PR?

No, only for the vara-cs stuff.

**Review by boehmseb — APPROVED** [review]

**Pull request merged**